### PR TITLE
[1.18] Add container out of memory metrics

### PR DESCRIPTION
**Pull request by discordianfish**
Backport of #4690 to 1.18

----
This patch adds a new metric
`container_runtime_crio_containers_oom_total` and
`container_runtime_crio_containers_oom`, which collects out of memory
(oom) metrics global and by container name. This also includes
sandboxes, since we reuse the CRI-O internal name for every container.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

```release-note
Added `container_runtime_crio_containers_oom_total` and  `container_runtime_crio_containers_oom` metrics,
which collects out of memory (oom) containers.
```

### CODE REVIEW

**Comment by haircommander** [comment]
/retitle [1.18] Add container out of memory metrics

/ok-to-test

**Comment by haircommander** [comment]
context: https://kubernetes.slack.com/archives/CAZH62UR1/p1638357795128600

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -94,6 +100,25 @@ var (
...
+			Help:      "Amount of containers killed because they ran out of memory (OOM)",
+		},
+	)
+
+	// CRIOContainersOOM collects container out of memory (oom) metrics per container and sandbox name.
+	CRIOContainersOOM = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: subsystem,
+			Name:      CRIOContainersOOMKey,
+			Help:      "Amount of containers killed because they ran out of memory (OOM) by their name",
```
Comment (TomSweeneyRedHat) [inline server/metrics/metrics.go:118]: Is this really the "Amount" or the "The names"?

Comment (discordianfish) [reply server/metrics/metrics.go:118]: I've just backported #4690 here, but I already filled #5426 to rework the metrics in general.

**Comment by haircommander** [comment]
all commits but the first can be dropped as the other 1.18 PR merged :slightly_smiling_face:

**Pull request merged**